doc/contributing/docs/markup/code.rst

```diff
@@ -1,8 +1,49 @@
 Writing about code
 ==================
 
+When writing articles, you have to format code specially, separating it from
+other text. This document will guide you through typical cases when
+it is recommended to use code highlighting.
+
+Defining what code is
+---------------------
+
+It is not difficult to recognize the code snippet in the text. First of all,
+it is probably code if the expression contains characters that ordinary words
+do not have, such as ``_, {}, [ ], .``.
+Also, you should format the expression as code if it fits at least one
+of the items in the list below:
+
+*   attribute names, values and class names,
+*   constant values for attributes,
+*   method and function names,
+*   language keywords,
+*   data types,
+*   error messages,
+*   URL for localhost and intranet,
+*   port number,
+*   package names,
+*   CLI app names.
+
+**Items we don't format as code**
+
+*   names of products, organizations and services, for example, Tarantool,
+    memtx, vinyl
+*   well-established terms such as stdin and stdout
+
+Code-blocks and inline code
+---------------------------
+
+If you have to choose between inline code and code-block highlighting,
+pay attention to the following guidelines:
+
 Code snippets
--------------
+~~~~~~~~~~~~~
+Use code-blocks when you have to highlight multiple lines of code.
+Also, use it if your code snippet contains a standalone element
+that is not a part of the article's text. In code-blocks, you can enable
```

Comment:
There are two separate things:

* (1) A code block, one of block elements on a page, the other being paragraph, list, table, admonition block, and so on. It can have a plural form: "code blocks". It exists on web and in books, and is 
* (2) The `code-block` directive in reStructuredText, used to make code blocks (1). There's no `code-blocks` directive and the plural form is "`code-block` directives". 

This section is about choosing between inline code and code blocks (1).

So, this can be an example for one more good idea: grammar doesn't apply to code, even inline.

* Correct: "use `shellcheck` to analyze your Bash code".
* Incorrect: "`shellcheck` your Bash code". It's best to not use code as a verb.
* Even worse: "shellcheck your Bash code". There's no such word in English and we don't explain what to use.
* Cursed: "try `shellchecking` your Bash code". There's no such word and no such application.

Please add this to the document.